addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -0,0 +1,390 @@
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the wso2iots handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            if (channelUID.getId().equals(wso2iotsBindingConstants.TEMPERATURE)) {
+                updateChannel(channelUID.getId(), iotsResponseTemp);
+            }
+            if (channelUID.getId().equals(wso2iotsBindingConstants.LIGHT)) {
+                updateChannel(channelUID.getId(), iotsResponseLight);
+            }
+            if (channelUID.getId().equals(wso2iotsBindingConstants.MOTION)) {
+                updateChannel(channelUID.getId(), iotsResponseMotion);
+            }
+            if (channelUID.getId().equals(wso2iotsBindingConstants.HUMIDITY)) {
+                updateChannel(channelUID.getId(), iotsResponseHumidity);
+            }
+        } else {
+            logger.debug("The Building Monitor binding is read-only and can not handle command {}", command);
+        }
+    }
+
+    /**
+     * Start the job refreshing the building monitor data
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request new data
+                        iotsResponseTemp = updateBuildingMonitorData(wso2iotsBindingConstants.TEMPERATURE);
+                        iotsResponseLight = updateBuildingMonitorData(wso2iotsBindingConstants.LIGHT);
+                        iotsResponseMotion = updateBuildingMonitorData(wso2iotsBindingConstants.MOTION);
+                        iotsResponseHumidity = updateBuildingMonitorData(wso2iotsBindingConstants.HUMIDITY);
+
+                        // Update all channels from the updated data
+                        updateChannel(wso2iotsBindingConstants.TEMPERATURE, iotsResponseTemp);
+                        updateChannel(wso2iotsBindingConstants.LIGHT, iotsResponseLight);
+                        updateChannel(wso2iotsBindingConstants.MOTION, iotsResponseMotion);
+                        updateChannel(wso2iotsBindingConstants.HUMIDITY, iotsResponseHumidity);
+
+                        // for (Channel channel : getThing().getChannels()) {
+                        // updateChannel(channel.getUID().getId(), iotsResponse);
+                        // }
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
```

Comment:
You can store the parameters locally, so you don't parse the config to the class at each refresh. When new configuration appears, the binding will be disposed and initialized again.

Reply:
I thought if binding is not initialized every time the configurations change and if a long time period is given as the refresh interval (ex: 1 hour), then the user wont be able to know until 1 hour whether the data is received properly after configuration is changed. Therefore is it better to initialize the binding each time the configurations change?

Reply:
The framework will call initialize after each configuration change for you